registrations/test_views.py

```diff
@@ -235,6 +235,69 @@ def test_get_registration_only_by_user(self):
         self.assertEqual(response.status_code, 403)
 
 
+class FacilityCodeCheckViewTests(AuthenticatedAPITestCase):
+    @responses.activate
+    @override_settings(JEMBI_BASE_URL="http://jembi/ws/rest/v1/")
+    def test_facility_code_check(self):
+
+        """
+            Test on Facility Code Check
+            GET - returns if facility code is correct, else return 200 response
+        """
+        self.make_source_normaluser()
+
+        clinic_code = 123456
+
+        result = {
+            "title": "Facility Code Check",
+            "headers": [],
+            "rows": [[clinic_code, "abcdefg", "test facility code"]],
+            "width": 1,
+            "height": 1,
+        }
+        responses.add(
+            responses.GET,
+            "http://jembi/ws/rest/v1/facilityCheck?{}".format(
+                urlencode({"criteria": "value:{}".format(clinic_code)})
+            ),
+            json=result,
+            status=200,
+        )
+        url = "{}?{}".format(
+            reverse("facilitycode-check"), urlencode({"clinic_code": clinic_code})
+        )
+        response = self.normalclient.get(url)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(json.loads(response.content)["Facility"], "test facility code")
+
+    @responses.activate
+    @override_settings(JEMBI_BASE_URL="http://jembi/ws/rest/v1/")
+    def test_facility_code_check_fail(self):
+
+        """
+            Test on Facility Code Check
+            GET - returns 400 response if facility code is incorrect
+        """
+
+        clinic_code = 111111
+
+        self.make_source_normaluser()
+        result = {"request_error": "HTTP 400 Bad Request"}
+        responses.add(
+            responses.GET,
+            "http://jembi/ws/rest/v1/facilityCheck?{}".format(
+                urlencode({"criteria": "value:{}".format(clinic_code)})
+            ),
+            json=result,
+            status=400,
+        )
+        url = "{}?{}".format(
+            reverse("facilitycode-check"), urlencode({"clinic_code": clinic_code})
+        )
+        response = self.normalclient.get(url)
+        self.assertEqual(response.status_code, 400)
```

Comment:
I think we also need a test for if there's no clinic for the given code, eg. someone types in the code wrong. In that case Jembi's API will just return an empty array. We handle it here in the jsbox: https://github.com/praekeltfoundation/ndoh-jsbox/blob/develop/src/ussd_clinic.js#L214-L215